### ISSUE TITLE
Function signature mismatch for `functools.reduce` between C implementation, Python implementation, and online documentaion

### ISSUE THREAD

**Issue by XuehaiPan** [python/cpython#102757]
# Documentation

(A clear and concise description of the issue.)

The function signature mismatch for `functools.reduce`:

- C implementation:

    ```python
    _initial_missing = object()

    @overload
    def reduce(function, iterable, /): ...
    @overload
    def reduce(function, iterable, initial=_initial_missing, /): ...
    ```

- Python implementation:

    ```python
    _initial_missing = object()

    @overload
    def reduce(function, sequence): ...
    @overload
    def reduce(function, sequence, initial=_initial_missing): ...
    ```

Argument change: `iterable -> sequence`.

- Online documentation:

    ```python
    def reduce(function, iterable, initializer=None): ...
    ```

Argument change: `initial -> initializer`.

### Linked PRs
* gh-102759

**Comment by XuehaiPan** [comment]
Can anyone have a look at this and my PR #102759?

**Comment by XuehaiPan** [comment]
Closed via PR #102759.
